docs/389ds/FAQ/bdb-deprecation.md

```diff
@@ -0,0 +1,277 @@
+
+Since 389-ds-base 2.1.0, The 389 ldap directory server supports two kind of underlying database:
+
+- Berkeley Database (bdb)
+
+- Lightning Memory-Mapped Database Manager (lmdb)
+
+Starting from version 3.0.0 New instances are no more created with bdb by default but with libdb.
+
+Newly created instances are still created with bdb by default while libdb is flagged as [deprecated since Fedora 33]([Changes/Libdb deprecated - Fedora Project Wiki](https://fedoraproject.org/wiki/Changes/Libdb_deprecated)), this change is about to create lmdb instances by default.
+
+{% include toc.md %}
+
+## Why deprecating bdb backend ?
+
+Because the underlying library implementing the bdb version used by 389-ds-base is no more supported upstream and it is already deprecated in some OS:
+
+- [Changes/Libdb deprecated - Fedora Project Wiki](https://fedoraproject.org/wiki/Changes/Libdb_deprecated)
+
+## How to configure a new instance with lmdb or bdb ?
+
+There are several ways:
+
+1. Using dscreate --interactive:
+   Select the proper implementation when the following question is asked:
+
+   ```
+   Choose whether mdb or bdb is used.
+   ```
+
+2. Using dscreate --from-file
+   Set the following stanza in the template file:
+
+   ```
+   [slapd]
+   db_lib = mdb (or bdb)
+   ```
+
+3. Using dscreate with an environment variable.
+   Setting NSSLAPD_DB_LIB environment variable to 'bdb' or 'mdb' change the backend implementation chosen by default
+
+Set  db_lib = mdb (or bdb) in [slapd] section of dscreate template file
+
+## Can I mix different implementations ?
+
+Mixing implementaions limitations are:
+
+- The db_lib parameter is stored in the nsslapd-backend-implement attribute of cn=config entry, so it is global to the instance.
+- backup / restore formats are tied to the db implementation so you cannot use a backup to restore an instance with a different implementation. (ldif export/import and initializing through the replication have not this limitation)
+
+| Question                                                                    | Answer |
+| --------------------------------------------------------------------------- | ------ |
+| Can I mix backends with different implementations in an instance ?          | No     |
+| Can I mix instances with different backends implementations in an host      | Yes    |
+| Can I mix replicas with different implementation in a replicated topology ? | Yes    |
+
+## How to migrate between database implementation ?
+
+There are several ways to do that.
+
+### Automatic method
+
+- dsctl *instanceName* dblib bdb2mdb
+
+- dsctl *instanceName* dblib mdb2bdb
+
+Note: these tools does not cleanup the old data (so can revert by simply changing nsslapd-backend-implement back).
+To remove the old database, after verifying that everything is working as expected, the following command should be run:
+
+- dsctl *instanceName* dblib cleanup
+
+#### Manual Method - export to ldif
+
+[A] Export all the backends to ldif
```

Comment:
Should we say suffixes and sub-suffixes ?
Just to eliminate doubt about special backends (cn=config, cn=schema, cn=monitor,..)

Reply:
I do not think there is any ambiguity here as there is no way to export special backends.

Reply:
Yes that is correct. forget my question